src/script/sconsign.py

```diff
@@ -487,6 +489,9 @@ def __call__(self, fname):
         except Exception as e:
             sys.stderr.write("sconsign: ignoring invalid `%s' file `%s': %s\n"
                              % (self.dbm_name, fname, e))
+            exc_type, _, _ = sys.exc_info()
+            if exc_type.__name__ == "ValueError" and sys.version_info < (3,0,0):
```

Comment:
Can you do if e is ValueError ?

Reply:
Apparently not... just tried it.

Reply:
Was curious... turns out we can't test if e is ValueError, but we can to the same test doing type. That is, this model would apparently work:

```
except Exception as e:
    # standard "ignore" message
    if type(e) == ValueError:
        # retrieve sys.exc_info and format message
```

I can submit a further change if you'd prefer the code to look this way.